tests/php/includes/TestUtils.php

```diff
@@ -0,0 +1,103 @@
+<?php
+/**
+ * Test the includes/utils.php functionality
+ *
+ * @package elasticpress
+ */
+
+namespace ElasticPressTest\Utils;
+
+use ElasticPress;
+
+/**
+ * Stats test class
+ */
+class TestUtils extends \ElasticPressTest\BaseTestCase {
+
+	/**
+	 * Setup each test.
+	 *
+	 * @since 3.2
+	 */
+	public function setUp() {
+		global $wpdb;
+		parent::setUp();
+	}
+
+	/**
+	 * Clean up after each test.
+	 *
+	 * @since 3.2
+	 */
+	public function tearDown() {
+		parent::tearDown();
+	}
+
+	/**
+	 * Test the flatten_tree_array function.
+	 *
+	 * @return void
+	 */
+	public function test_flatten_tree_array() {
+		$this->generate_test_flatten_tree_array( 0, 0 );
+		$this->generate_test_flatten_tree_array( 1, 1 );
+		$this->generate_test_flatten_tree_array( 3, 4 );
+		$this->generate_test_flatten_tree_array( 100, 5 );
+		$this->generate_test_flatten_tree_array( 5, 100 );
+		$this->generate_test_flatten_tree_array( 5, 0 );
+	}
+
+	/**
+	 * Generate test function for testing flatten_tree_array function.˝
+	 *
+	 * @param integer $base_size Array tree parent size.
+	 * @param integer $child_size Array tree parent child size.
+	 * @return void
+	 */
+	private function generate_test_flatten_tree_array( $base_size = 1, $child_size = 1 ) {
+
+		$expected_size = ( $base_size ) + ( $base_size ) * ( $child_size );
+
+		$test_object = $this->generate_array_tree( $base_size, $child_size );
+
+		$flatten_array = ( \ElasticPress\Utils\flatten_tree_array( $test_object ) );
+
+		$this->assertEquals( $expected_size, count( $flatten_array ) );
+
+	}
```

Comment:
I think this might work better with a [dataProvider](https://phpunit.de/manual/3.7/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers)

(I haven't tested the suggestion bellow, just to ilustrate what I meant. Feel free to refactor)

```suggestion
   public function test_flatten_tree_array_data() {
		return [
		 [1,1],
		 [3,4],
		 ...
		];
   }

	/**
     * @dataProvider test_flatten_tree_array_data
	 */
	private function test_flatten_tree_array( $base_size = 1, $child_size = 1 ) {

		$expected_size = ( $base_size ) + ( $base_size ) * ( $child_size );

		$test_object = $this->generate_array_tree( $base_size, $child_size );

		$flatten_array = ( \ElasticPress\Utils\flatten_tree_array( $test_object ) );

		$this->assertEquals( $expected_size, count( $flatten_array ) );

	}
```